components/provider_ppucd/src/parse_ppucd.rs

```diff
@@ -0,0 +1,450 @@
+            // Properties with an "=" in their string have values associated
+            // (ex: enumerated properties).
+            m.insert(&prop_str[0..idx], &prop_str[idx + 1..]);
+        } else {
+            // For properties that don't take values, let their value in the map be the prop name itself
+            // This applies to binary properties.
+            m.insert(prop_str, prop_str);
+        }
+    }
+    m
+}
+
+fn is_skip_ppucd_line(line: &str) -> bool {
+    line.starts_with('#') || line.starts_with("ucd") || line.is_empty()
+}
+
+fn is_property_line(line: &str) -> bool {
+    line.starts_with("property;")
+}
+
+/// For a property definition line, update the property aliases map.
+/// Only operate on binary properties, currently.
+fn update_aliases<'s>(prop_aliases: &mut HashMap<&'s str, HashSet<&'s str>>, line: &'s str) {
+    let mut line_parts = split_line(&line);
+    assert_eq!(&"property", &line_parts[0]);
+    let prop_type = &line_parts[1];
+    if prop_type == &"Binary" {
+        line_parts.drain(0..2);
+
+        // TODO: ask Markus what to do with the property lines that appear to have
+        // no canonical name
+        // property;Binary;;alnum
+        // property;Binary;;blank
+        // property;Binary;;graph
+        // property;Binary;;print
+        // property;Binary;;xdigit
+        if line_parts[0].is_empty() {
+            line_parts.drain(0..1);
+        }
+
+        let canonical_name = line_parts[0];
+        let all_names: Vec<&'s str> = line_parts.iter().copied().collect();
+        let all_names_set: HashSet<&'s str> = all_names.into_iter().collect();
+        prop_aliases.insert(canonical_name, all_names_set);
+    }
+}
+
+fn is_defaults_line(line: &str) -> bool {
+    line.starts_with("defaults;")
+}
+
+/// Return the property key-value information represented in the "defaults"
+/// line as a map. "defaults" is like the base level of overrides of property
+/// values for all code points in PPUCD.
+fn get_defaults_prop_vals(line: &str) -> HashMap<&str, &str> {
+    let line_parts = split_line(&line);
+    assert_eq!(&"defaults", &line_parts[0]);
+    get_data_line_prop_vals(&line_parts)
+}
+
+fn is_block_line(line: &str) -> bool {
+    line.starts_with("block;")
+}
+
+/// Return the property key-value information represented in a "blk"
+/// line as a map. "blocks" represent overrides of property values for code
+/// points in PPUCD within a Unicode block above the "defaults" values.
+fn get_block_range_prop_vals(line: &str) -> (UnicodeSet, HashMap<&str, &str>) {
+    let line_parts = split_line(&line);
+    assert_eq!(&"block", &line_parts[0]);
+
+    let range_str = &line_parts[1];
+    let range_bound_strs = range_str.split("..").collect::<Vec<_>>();
+    let range_start: &u32 = &u32::from_str_radix(&range_bound_strs[0], 16).unwrap();
+    let range_end: &u32 = &u32::from_str_radix(&range_bound_strs[1], 16).unwrap(); // inclusive end val in PPUCD
```

Comment:
We're trying to avoid `unwraps` - can we either return `Result` from here and handle it in the bin, or at least `expect`?

Reply:
I agree, but I think it's okay to allow unwraps in provider_ppucd for the time being since it is intended to be run primarily at build time rather than runtime.